### DISCUSSION TITLE
Say Hello!

### DISCUSSION THREAD

**Discussion by bmcollier** [bmcollier/lineage#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dconlan** [comment]
Hi Ben,
  I've been working with this code on and off for the last few years (but only doing tweaks), how does an animal pedigree differ from human? I'm curious about what sort of changes you are looking at making.

David Conlan

**Reply by bmcollier** [reply]
Hi David, the focus for me at the moment is really on providing tools for animal breeders, where instead of generating a pedigree and storing information for a single mutation of interest, we want to track up to the order of 100 mutations simultaneously, and be able to switch between selected mutations in the main visualisation.

Further to that, there are a number of measures of animal health from statistical genetics which we can generate on the fly. Elements like the Proband will need looking at to decide how they're handled in this context as well.

I'll also probably add some functionality to save their pedigrees to the cloud and link them to one another - this will be useful for breeding organisations.

The statistical genetics stuff I can contribute back to the original project, as well as a number of changes and fixes to the graphical elements to support larger more complex trees with greater levels of inbreeding.

**Reply by bmcollier** [reply]
Oh, I should add that there are some fairly obvious changes that I'll make to the general data sets associated with individuals on a pedigree. First name, last name doesn't make sense for animals, and there are a number of other things that are of interest and worth tracking like weights and diet.

Really the priority of all of this stuff will depend on take-up of the tool. Right now I'm just adding the stuff that's useful to me.

**Reply by dconlan** [reply]
Sounds like it could be a fair bit of work. I imagine with animals, you could be looking at traits for breeding purposes, or diseases for medical purposes. 
I know an option was added to remove a page of the person node as it wasn't of interest. There is also a number of fields which could be relevant to associate with a gene (variance, zygosity). 

What we need is some form of templating mechanism, where you define the data to be attached to a 'person' node. So the template would define the set of attributes and there types that would come up in dialog that currently shows name and dob. Ultimately you would want some way for an attribute to effect how the node is rendered.

**Reply by bmcollier** [reply]
Yes, I think you're right about the templating. The two bits which I'd like to be able to abstract in this way are the node data elements, and the mapping to types of export. This would make things a lot easier to work with, and then arguably we wouldn't need the fork at all. As regards export / import, there's a strategy pattern here that could do with being implemented as well, to enable support for other data formats.

**Reply by bmcollier** [reply]
In terms of the immediate roadmap for this fork, I'm going to change the base system to support a basic animal genetics function (mapping name and first name to other elements and changing the menu a bit), then implement a custom load and save functionality, so I can get the thing online and working for multiple users.

I think after that we may be in refactoring territory to support the extensibility features as previously discussed.
